### DISCUSSION TITLE
Is there an easy to follow example of an Express BE + NextJS FE?

### DISCUSSION THREAD

**Discussion by Randulfe** [nextauthjs/next-auth#12306] [Help] [answered]
I've got my Express API that manages all DB connections on the BE and a NextJS client where ideally I would like to be able to recycle those sign in buttons and pages. Is there an easy way to achieve this?

Constrains: 
- The client NextJS side doesn't have access to the database or its schema, only the BE express server has 
- The FE and BE live in different domains and I dont want to expose the BE domain to the public

**Comment by Randulfe** [comment, marked as answer]
Ok for whoever gets here, this is how you sort some of the issues with the integration: 
Your FE client: 
- First make a request to get the CSRF token 
- Then make a POST request to your Express auth endpoint: `domain/auth/signin/[provider]`
    - Make sure that you are including credentials `withCredentials: true` in axios or `credentials: 'include'` in vanilla fetch 
- This request will respond with a corresponding URL that you want to redirect the user to so that they can sign in with the provider

Your express BE: 
- Make sure when defining the `ExpressAuth` you are setting the property `callbacks: async redirect() { return [the URL of your home signed in page in your FE] }`. This will redirect the application to your client's site instead of to an express endpoint.

**Reply by skev007** [reply]
hey, did you manage to successfully complete the integration for this setup? 
if yes, can you share the example code for handling the sessions and auth in the frontend and how are you handling the CORS between your frontend and backend (assuming both are running on separate domains/ports)
